dcp-0006/dcp-0006.mediawiki

```diff
@@ -0,0 +1,392 @@
+* There shall be one input script.
+* There shall be at least two output scripts.
+* All output scripts shall be version 0.
+* TxIn[0] signature script shall be exactly 100 bytes in length. The script format is <code>OP_DATA_64</code> (signature), <code>OP_DATA_33</code> (Pi public key) followed by an <code>OP_TSPEND</code>.
+** The Pi public key that is included in TxIn[0] shall be of the strict compressed public key encoding type.
+* TxOut[0] script shall be an <code>OP_RETURN</code> <code>OP_DATA_32</code>.
+* TxOut[1:N] scripts shall be an <code>OP_TGEN</code> followed by a P2PKH or a P2SH script.
+
+{|
+!Transaction!!Script!!Description
+|-
+|TxIn[0]||[0]OP_DATA_64 signature<br>[65]OP_DATA_33 Pi public key<br>[99]OP_TSPEND||The value shall be sum(expenditures) + mining fee.
+|-
+|TxOut[0]||[0]OP_RETURN<br>[1]OP_DATA_32||Bytes that make the entire transaction unique. First 8 bytes are the little endian encoded value from TxIn[0]. The following 24 bytes are random.
+|-
+|TxOut[1:N]||[0]OP_TGEN<br>[1:M]P2PKH or P2SH script||These are the individual payouts from the treasury account.
+|}
+
+Additional consensus enforced characteristics:
+* The previous out point for TxIn[0] shall be null.
+* The TSpend transaction fraud proof shall be null.
+* The length of the signature script in TxIn[0] shall be 100 bytes ([OP_DATA_64] [signature] [OP_DATA_33] [PiKey] [OP_TSPEND] = 1 + 64 + 1 + 33 + 1 = 100)
+* A TSpend transaction shall not be allowed in a block that is not a TVI.
+* A TSpend transaction shall not be allowed in a block if the start of the voting window is prior to stake validation height.
+* A TSpend transaction shall not be allowed in a block if the Expiry is outside of the TSpend validity window.
+* The value of input 0 of a TSpend transaction shall be int64 little endian encoded in the output 0 script in bytes 2 through 10. The value of both shall be identical.
+* A TSpend transaction shall not be allowed in a block if it has been mined in a prior block (this is to enable support for short-circuiting a vote) that is in the chain of ancestors.
+* A TSpend transaction shall not be allowed in a block if it does not reach quorum which is calculated as follows: (sum(yes votes) + sum(no votes)) * TVQM / TVQD
+* A TSpend transaction shall not be allowed in a block if it does not reach enough yes votes  which is calculated as follows: (sum(votes cast) + sum(possible remaining votes)) * TVRM / TVRD
+* A TSpend transaction shall not deplete the entire treasury. The treasury balance shall not become negative.
+* The sum of all TSpend transactions within the most recent TPE shall not exceed 150% of avg(sum(tspends), TWE).
+* A TSpend transaction shall not be allowed in a block if the secp256k1 public key is not well-known (encoded in chain parameters).
+* A TSpend transaction shall not be allowed in a block if the Schnorr signature of the transaction does not match the provided well-known public key.
```

Comment:
We need to include somewhere that the sig hash type used is `SigHashAll`.

Reply:
Dup